src/blueprint.zig

```diff
@@ -274,6 +277,64 @@ pub const Blueprint = struct {
 			},
 		}
 	}
+	pub fn set(self: Blueprint, allocator: NeverFailingAllocator, pattern: Pattern) Blueprint {
+		var copy = self.clone(allocator);
```

Comment:
In my opinion the caller should always be responsible of copying, and set should take a variable pointer.
Otherwise it's easy to make additional copies when you don't actually need them (e.g. when /set is applied to the clipboard).

Furthermore cloning is not necessary here anyways, you overwrite all values afterwards anyways, so you just need to create a new blueprint of identical size.

Reply:
> In my opinion the caller should always be responsible of copying, and set should take a variable pointer. Otherwise it's easy to make additional copies when you don't actually need them (e.g. when /set is applied to the clipboard).

That is true, I will change that.

> Furthermore cloning is not necessary here anyways, you overwrite all values afterwards anyways, so you just need to create a new blueprint of identical size.

That is true only because there are no masks yet, after masks are implemented it will be possible that some (or even all) of old blocks will be preserved.

Reply:
Changed first, left second, I will add mask parameter to `fn set` after #1284